swift-mode-font-lock.el

```diff
@@ -54,181 +54,341 @@ Return nil otherwise."
-    ("\\<print\\>" . font-lock-builtin-face)
-    ("\\<readLine\\>" . font-lock-builtin-face)
-    ("\\<sizeof\\>" . font-lock-builtin-face)
-    ("\\<sizeofValue\\>" . font-lock-builtin-face)
-    ("\\<strideof\\>" . font-lock-builtin-face)
-    ("\\<strideofValue\\>" . font-lock-builtin-face)
-    ("\\<swap\\>" . font-lock-builtin-face)
-    ("\\<transcode\\>" . font-lock-builtin-face)
-    ("\\<unsafeAddressOf\\>" . font-lock-builtin-face)
-    ("\\<unsafeBitCast\\>" . font-lock-builtin-face)
-    ("\\<unsafeDowncast\\>" . font-lock-builtin-face)
-    ("\\<unsafeUnwrap\\>" . font-lock-builtin-face)
-    ("\\<withExtendedLifetime\\>" . font-lock-builtin-face)
-    ("\\<withUnsafeMutablePointer\\>" . font-lock-builtin-face)
-    ("\\<withUnsafeMutablePointers\\>" . font-lock-builtin-face)
-    ("\\<withUnsafePointer\\>" . font-lock-builtin-face)
-    ("\\<withUnsafePointers\\>" . font-lock-builtin-face)
-    ("\\<withVaList\\>" . font-lock-builtin-face)
-    ("\\<zip\\>" . font-lock-builtin-face)
-
-    ;; keywords for build configuration statements
-    ("\\<os\\>" . font-lock-builtin-face)
-    ("\\<arch\\>" . font-lock-builtin-face)
-    ("\\<swift\\>" . font-lock-builtin-face)
-    ("\\<OSX\\>" . font-lock-builtin-face)
-    ("\\<iOS\\>" . font-lock-builtin-face)
-    ("\\<watchOS\\>" . font-lock-builtin-face)
-    ("\\<tvOS\\>" . font-lock-builtin-face)
-    ("\\<i386\\>" . font-lock-builtin-face)
-    ("\\<x86_64\\>" . font-lock-builtin-face)
-    ("\\<arm\\>" . font-lock-builtin-face)
-    ("\\<arm64\\>" . font-lock-builtin-face)
-    ("\\<iOSApplicationExtension\\>" . font-lock-builtin-face)
-    ("\\<OSXApplicationExtension\\>" . font-lock-builtin-face)
-
-    (swift-mode:font-lock-match-function-names . font-lock-function-name-face)
-    )
+    (,(regexp-opt swift-mode:constant-keywords 'words)
+     .
+     font-lock-constant-face)
+
+    (,(regexp-opt swift-mode:preprocessor-keywords 'symbols)
+     .
+     font-lock-preprocessor-face)
+
+    ,(regexp-opt (append swift-mode:declaration-keywords
+                         swift-mode:statement-keywords
+                         swift-mode:expression-keywords
+                         swift-mode:context-keywords) 'words)
+
+    (,(concat "\\.\\s-*"
+              (regexp-opt swift-mode:member-functions-trailing-closure 'words)
+              "\\s-*[({]")
+     1
+     font-lock-builtin-face)
+
+    (,(concat "\\.\\s-*"
+              (regexp-opt swift-mode:member-functions-trailing-closure 'words)
+              "\\s-*(")
+     1
+     font-lock-builtin-face)
+
+    (,(concat (regexp-opt swift-mode:global-functions-trailing-closure 'words)
+              "\\s-*[({]")
+     1
+     font-lock-builtin-face)
+
+    (,(concat (regexp-opt swift-mode:global-functions 'words)
+              "\\s-*(")
+     1
+     font-lock-builtin-face)
+
+    (,(concat "\\(^Process\\.\\|[^_[:alnum:]]Process\\.\\|^CommandLine\\.\\|[^_[:alnum:]]CommandLine\\.\\)"
```

Comment:
Why not just `\\<...\\>`? Should `\\s-*` be inserted around the dot?

Reply:
Hmm, good point. This only matches fields pertaining to `Process`/`CommandLine`, but I suppose that would be simpler. I've updated it to be combined with the other properties.